### PR TITLE
NF-589 - Refine NDRC CYA page when answers are missing

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/utils/RepaymentAmountSummaryAnswersHelper.scala [hunk 1]
```diff
@@ -123,14 +123,14 @@ class RepaymentAmountSummaryAnswersHelper(userAnswers: UserAnswers)(implicit mes
   private def getAnswerSection(dutyType: String, dutyPaid: Double, dutyDue: Double): AnswerSection = {
     val answerSection: AnswerSection = dutyType match {
       case "repaymentAmountSummary.total" =>
-        AnswerSection(Some(dutyType), Seq(displayDuty("0", dutyPaid, "repaymentAmountSummary.total.amount").get))
+        AnswerSection(Some(dutyType), Seq(displayDuty("0", dutyPaid, "repaymentAmountSummary.total.amount")))
```
Comment (SteveSugden) [inline app/utils/RepaymentAmountSummaryAnswersHelper.scala:126]: Found these problems when doing some "extreme" testing of missing answers (i.e. all answers missing).  Better to render an empty row than to blow up I think.

File: app/utils/RepaymentAmountSummaryAnswersHelper.scala [hunk 2]
```diff
@@ -29,8 +29,8 @@ class RepaymentAmountSummaryAnswersHelper(userAnswers: UserAnswers)(implicit mes
     def format2d = "%.2f".format(s)
   }
 
-  def displayDuty(index: String, amount: Double, dutyType: String): Option[AnswerRow] =
-    getAnswerRow(index, amount, dutyType)
+  def displayDuty(index: String, amount: Double, dutyType: String): AnswerRow =
+    getAnswerRow(index, amount, dutyType).getOrElse(AnswerRow(Html(""), Html("")))
```
Comment (njmoorewitchcat) [inline app/utils/RepaymentAmountSummaryAnswersHelper.scala:33]: CLARIFICATION: Do these fixes in this file stop the Repayment Page throwing an exception?  If so, I have a bug report on the backlog that can be tied to this story and tested alongside.

**Review by njmoorewitchcat — APPROVED** [review]
LGTM

File: app/utils/RepaymentAmountSummaryAnswersHelper.scala (hunk 2 above)
Comment (SteveSugden) [reply app/utils/RepaymentAmountSummaryAnswersHelper.scala:33]: They do.   I remember you mentioning a repayment summary page bug yesterday.  I (re)discovered this when manually navigating to CYA with no answers in cache and trying to "change" the repayment amount.  In reality, I don't think there should be such an extreme example of missing answers.

**Pull request merged**